### DISCUSSION TITLE
exploring-the-dotnet-8-preview-changing-method-calls-with-interceptors/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#391] [Announcements]
# exploring-the-dotnet-8-preview-changing-method-calls-with-interceptors/

In this post I look at the C#12 feature, interceptors. I show how and why they're useful, and how the ASP.NET Core minimal API source generator uses them

https://andrewlock.net/exploring-the-dotnet-8-preview-changing-method-calls-with-interceptors/

**Comment by akhanalcs** [comment]
That was very interesting!

**Comment by julealgon** [comment]
Andrew, would it be possible to use this interception mechanism to intercept `lock` calls? Say, for example, I want to add logging every time a lock is acquired, and every time a lock is freed. Since `lock` is a keyword and not a method, I imagine it wouldn't work directly, but perhaps it could work if you intercept `Monitor.Enter` and `Monitor.Exit`? If so, how would I know the resulting line of code after the compiler transforms the `lock` call into the `Enter`/`Exit` pair?

**Reply by andrewlock** [reply]
I think the simple answer is you probably can't, but you could ask on the dotnet/runtime repo just in case!
